sktime/forecasting/hf_transformers_forecaster.py

```diff
@@ -359,35 +443,62 @@ def get_test_params(cls, parameter_set="default"):
 class PyTorchDataset(Dataset):
     """Dataset for use in sktime deep learning forecasters."""
 
-    def __init__(self, y, seq_len, fh=None, X=None):
-        self.y = y.values
-        self.X = X.values if X is not None else X
-        self.seq_len = seq_len
+    def __init__(self, y: pd.DataFrame, seq_len: int, fh=None, X: pd.DataFrame = None):
+        if not isinstance(y.index, pd.MultiIndex):
+            self.y = np.array(y.values, dtype=np.float32).reshape(len(y), 1)
+            self.X = (
+                np.array(X.values, dtype=np.float32).reshape(
+                    (len(X.columns), len(X), 1)
+                )
+                if X is not None
+                else X
+            )
+        else:
+            lens = y.groupby(level=list(range(len(y.index.levels) - 1))).apply(
```

Comment:
The code looks similar as in the _predict method. Could you check if it is possible to extract it in a method to reduce code complexity and duplication?

Reply:
Now numpy array in the `PyTorchDataset` are of the same shape as the array passed to predict, so the code is extracted  to a method and reused now.